recipes/numcpp/all/conanfile.py

```diff
@@ -0,0 +1,61 @@
+from conans import ConanFile, tools
+import os
+import glob
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.32.0"
+
+class NumCppConan(ConanFile):
+    name = "numcpp"
+    description = "A Templatized Header Only C++ Implementation of the Python NumPy Library"
+    topics = ("python", "numpy", "numeric")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/dpilger26/NumCpp"
+    license = "MIT"
+    no_copy_source = True
+    settings = "compiler"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def validate(self):
+        minimal_cpp_standard = "14"
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, minimal_cpp_standard)
+        minimal_version = {
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "10",
+            "Visual Studio": "14"
+        }
+        compiler = str(self.settings.compiler)
+        if compiler not in minimal_version:
+            self.output.warn(
+                "%s recipe lacks information about the %s compiler standard version support" % (self.name, compiler))
+            self.output.warn(
+                "%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+            return
+        version = tools.Version(self.settings.compiler.version)
+        if version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration("%s requires a compiler that supports at least C++%s" % (self.name, minimal_cpp_standard))
+
+    def requirements(self):
+        self.requires("boost/1.75.0")
```

Comment:
which features of boost are used? Any boost components checks to do in validate() ? if it depends on boost header only or specific components, it can also be enforced in `packageç_info()` to avoid to propagate the all boost libraries in generators.

Reply:
```
./include/NumCpp/Functions/gcd.hpp:37:#include "boost/integer/common_factor_rt.hpp"
./include/NumCpp/Functions/lcm.hpp:37:#include "boost/integer/common_factor_rt.hpp"
./include/NumCpp/Polynomial/chebyshev_t.hpp:36:#include "boost/math/special_functions/chebyshev.hpp"
./include/NumCpp/Polynomial/chebyshev_u.hpp:36:#include "boost/math/special_functions/chebyshev.hpp"
./include/NumCpp/Polynomial/hermite.hpp:39:#include "boost/math/special_functions/hermite.hpp"
./include/NumCpp/Polynomial/laguerre.hpp:39:#include "boost/math/special_functions/laguerre.hpp"
./include/NumCpp/Polynomial/legendre_p.hpp:40:#include "boost/math/special_functions/legendre.hpp"
./include/NumCpp/Polynomial/legendre_q.hpp:36:#include "boost/math/special_functions/legendre.hpp"
./include/NumCpp/Polynomial/spherical_harmonic.hpp:35:#include "boost/math/special_functions/spherical_harmonic.hpp"
./include/NumCpp/PythonInterface/BoostInterface.hpp:37:#include "boost/python.hpp"
./include/NumCpp/PythonInterface/BoostInterface.hpp:38:#include "boost/python/numpy.hpp"
./include/NumCpp/PythonInterface/BoostNumpyNdarrayHelper.hpp:42:#include "boost/python.hpp"
./include/NumCpp/PythonInterface/BoostNumpyNdarrayHelper.hpp:43:#include "boost/python/numpy.hpp"
./include/NumCpp/Random/beta.hpp:38:#include "boost/random/beta_distribution.hpp"
./include/NumCpp/Random/laplace.hpp:37:#include "boost/random/laplace_distribution.hpp"
./include/NumCpp/Random/nonCentralChiSquared.hpp:38:#include "boost/random/non_central_chi_squared_distribution.hpp"
./include/NumCpp/Random/triangle.hpp:39:#include "boost/random/triangle_distribution.hpp"
./include/NumCpp/Random/uniformOnSphere.hpp:39:#include "boost/random/uniform_on_sphere.hpp"
./include/NumCpp/Special/airy_ai.hpp:36:#include "boost/math/special_functions/airy.hpp"
./include/NumCpp/Special/airy_ai_prime.hpp:36:#include "boost/math/special_functions/airy.hpp"
./include/NumCpp/Special/airy_bi.hpp:36:#include "boost/math/special_functions/airy.hpp"
./include/NumCpp/Special/airy_bi_prime.hpp:36:#include "boost/math/special_functions/airy.hpp"
./include/NumCpp/Special/bernoulli.hpp:36:#include "boost/math/special_functions/bernoulli.hpp"
./include/NumCpp/Special/bessel_in.hpp:39:#include "boost/math/special_functions/bessel.hpp"
./include/NumCpp/Special/bessel_in_prime.hpp:36:#include "boost/math/special_functions/bessel_prime.hpp"
./include/NumCpp/Special/bessel_jn.hpp:39:#include "boost/math/special_functions/bessel.hpp"
./include/NumCpp/Special/bessel_jn_prime.hpp:36:#include "boost/math/special_functions/bessel_prime.hpp"
./include/NumCpp/Special/bessel_kn.hpp:39:#include "boost/math/special_functions/bessel.hpp"
./include/NumCpp/Special/bessel_kn_prime.hpp:36:#include "boost/math/special_functions/bessel_prime.hpp"
./include/NumCpp/Special/bessel_yn.hpp:39:#include "boost/math/special_functions/bessel.hpp"
./include/NumCpp/Special/bessel_yn_prime.hpp:36:#include "boost/math/special_functions/bessel.hpp"
./include/NumCpp/Special/beta.hpp:39:#include "boost/math/special_functions/beta.hpp"
./include/NumCpp/Special/comp_ellint_1.hpp:39:#include "boost/math/special_functions/ellint_1.hpp"
./include/NumCpp/Special/comp_ellint_2.hpp:39:#include "boost/math/special_functions/ellint_2.hpp"
./include/NumCpp/Special/comp_ellint_3.hpp:40:#include "boost/math/special_functions/ellint_3.hpp"
./include/NumCpp/Special/cyclic_hankel_1.hpp:35:#include "boost/math/special_functions/hankel.hpp"
./include/NumCpp/Special/cyclic_hankel_2.hpp:35:#include "boost/math/special_functions/hankel.hpp"
./include/NumCpp/Special/digamma.hpp:36:#include "boost/math/special_functions/digamma.hpp"
./include/NumCpp/Special/ellint_1.hpp:40:#include "boost/math/special_functions/ellint_1.hpp"
./include/NumCpp/Special/ellint_2.hpp:40:#include "boost/math/special_functions/ellint_2.hpp"
./include/NumCpp/Special/ellint_3.hpp:40:#include "boost/math/special_functions/ellint_3.hpp"
./include/NumCpp/Special/erf.hpp:36:#include "boost/math/special_functions/erf.hpp"
./include/NumCpp/Special/erfc.hpp:36:#include "boost/math/special_functions/erf.hpp"
./include/NumCpp/Special/erfc_inv.hpp:36:#include "boost/math/special_functions/erf.hpp"
./include/NumCpp/Special/erf_inv.hpp:36:#include "boost/math/special_functions/erf.hpp"
./include/NumCpp/Special/expint.hpp:39:#include "boost/math/special_functions/expint.hpp"
./include/NumCpp/Special/factorial.hpp:35:#include "boost/math/special_functions/factorials.hpp"
./include/NumCpp/Special/gamma.hpp:36:#include "boost/math/special_functions/gamma.hpp"
./include/NumCpp/Special/gamma1pm1.hpp:36:#include "boost/math/special_functions/gamma.hpp"
./include/NumCpp/Special/log_gamma.hpp:36:#include "boost/math/special_functions/gamma.hpp"
./include/NumCpp/Special/pnr.hpp:35:#include "boost/math/special_functions/factorials.hpp"
./include/NumCpp/Special/polygamma.hpp:36:#include "boost/math/special_functions/polygamma.hpp"
./include/NumCpp/Special/prime.hpp:37:#include "boost/math/special_functions/prime.hpp"
./include/NumCpp/Special/riemann_zeta.hpp:39:#include "boost/math/special_functions/zeta.hpp"
./include/NumCpp/Special/spherical_bessel_jn.hpp:39:#include "boost/math/special_functions/bessel.hpp"
./include/NumCpp/Special/spherical_bessel_yn.hpp:39:#include "boost/math/special_functions/bessel.hpp"
./include/NumCpp/Special/spherical_hankel_1.hpp:35:#include "boost/math/special_functions/hankel.hpp"
./include/NumCpp/Special/spherical_hankel_2.hpp:35:#include "boost/math/special_functions/hankel.hpp"
./include/NumCpp/Special/trigamma.hpp:36:#include "boost/math/special_functions/trigamma.hpp"
./include/NumCpp/Utils/powerf.hpp:33:#include "boost/type_traits/is_complex.hpp"
```